core/hotspot/rule_manager.go

```diff
@@ -219,6 +221,13 @@ func onRuleUpdate(rules []*Rule) (err error) {
 			}
 		}
 	}
+
+	tcMux.Lock()
+	defer func() {
```

Comment:
remove defer and move logging logic to the first defer

Reply:
done